.circleci/config.yml

```diff
@@ -175,11 +211,20 @@ workflows:
   version: 2
   workflow:
     jobs:
+#       This job will run on every commit on any branch
       - build_debug
-      - test_unit
-      - test_instrumented:
+#       Unit tests should only start after a successful build. Or vice versa?
```

Comment:
What are the trade offs. If it's something to try empirically, then might be easier to file an issue for a contributor to try.

Reply:
Two lines below the comment, there's another comment stating 
"#         Although the unit testing task could run in parallel with build task,
#         in such case, it would not benefit from the previous caching in the event of
#         any change in the .gradle script files.

For me that is enough to choose that unit tests depend on the build task. However, in case the other way is preferred, based on things I don't know yet i.e. organization policy (I've just jumped straight here out of nowhere ;), it's a trivial task to toggle. So, please advise.